library/Rules/Video.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+/*
+ * This file is part of Respect/Validation.
+ *
+ * (c) Alexandre Gomes Gaigalas <alexandre@gaigalas.net>
+ *
+ * For the full copyright and license information, please view the "LICENSE.md"
+ * file that was distributed with this source code.
+ */
+
+namespace Respect\Validation\Rules;
+
+class Video extends AbstractRule
+{
+    public function validate($input)
+    {
+        return !empty($input) && (
+            preg_match('/^.*((youtu.be\/)|(v\/)|(\/u\/\w\/)|(embed\/)|(watch\?))\??v?=?([^#\&\?]*).*/i', $input) ||
+            preg_match('/^.*(vimeo\.com\/)((channels\/[A-z]+\/)|(groups\/[A-z]+\/videos\/)|(video\/))?([0-9]+)/i', $input)
```

Comment:
I would probably go for only one rule, called `videoUrl()` which accepts argument on its construtor, that can be `"youtube"` or `"vimeo"`, them we can implement more, like `"ted"`, `"viki"`... You can create a key => value array with the patterns to match for the defined video provider (youtube, vimeo...) or try all of them if no provider is defined.